internal/api/provider/oidc.go

```diff
@@ -200,6 +204,64 @@ func parseLinkedinIDToken(token *oidc.IDToken) (*oidc.IDToken, *UserProvidedData
 	return token, &data, nil
 }
 
+type AzureIDTokenClaims struct {
+	jwt.StandardClaims
+
+	Email                              string `json:"email"`
+	XMicrosoftEmailDomainOwnerVerified *bool  `json:"xms_edov"`
+}
+
+func parseAzureIDToken(token *oidc.IDToken) (*oidc.IDToken, *UserProvidedData, error) {
+	var data UserProvidedData
+
+	var azureClaims AzureIDTokenClaims
+	if err := token.Claims(&azureClaims); err != nil {
+		return nil, nil, err
+	}
+
+	// If xms_edov is not set, and an email is present or xms_edov is true,
+	// only then is the email regarded as verified.
+	// https://learn.microsoft.com/en-us/azure/active-directory/develop/migrate-off-email-claim-authorization#using-the-xms_edov-optional-claim-to-determine-email-verification-status-and-migrate-users
+	emailVerified := false
+
+	if azureClaims.XMicrosoftEmailDomainOwnerVerified == nil {
+		// An email is provided, but xms_edov is not -- probably not
+		// configured, so we must assume the email is verified as Azure
+		// will only send out a potentially unverified email address in
+		// single-tenanat apps.
+		emailVerified = azureClaims.Email != ""
+	} else {
+		emailVerified = azureClaims.Email != "" && *azureClaims.XMicrosoftEmailDomainOwnerVerified
+	}
+
+	data.Metadata = &Claims{
+		Issuer:        token.Issuer,
+		Subject:       token.Subject,
+		Email:         azureClaims.Email,
+		EmailVerified: emailVerified,
+		ProviderId:    token.Subject,
+		CustomClaims:  make(map[string]any),
+	}
+
+	if data.Metadata.Email != "" {
+		data.Emails = append(data.Emails, Email{
+			Email:    data.Metadata.Email,
+			Verified: data.Metadata.EmailVerified,
+			Primary:  true,
+		})
+	}
+
+	if err := token.Claims(&data.Metadata.CustomClaims); err != nil {
```

Comment:
There seem to be a lot of nonstandard ones, and also a lot that can be customized so I'd actually just add all of them in as custom to allow flexibility to users.

Reply:
i would err on the side of not adding additional claims unless there is a usecase for them or else it just bloats the JWT returned

Reply:
I get that, but the use case is people adding their own claims in Azure. This is only available in Azure, you can't do that with Google or Apple -- so IMO this should be supported otherwise how else are they going to access the claims? I'll see about removing the standard claims that would be duplicated a bit.

Reply:
yeah that makes sense, but we haven't had anyone ask for specific custom claims from azure yet so i'm leaning on the side of not adding these claims until one of our users actually requests for it